### ISSUE TITLE
Autoplay not working with extractors

### ISSUE THREAD

**Issue by joshuayuen99** [Androz2091/discord-player#678]
**Describe the bug**
The autoplay feature just doesn't work at all.

**To Reproduce**
Play any song via any source (YouTube, Spotify, search phrase), and set the repeat mode to autoplay.

**Expected behavior**
A new song would play after the current song ends.

**Screenshots**

**Please complete the following information:**
 - Node Version: [16.6.1]
 - Discord Player Version: [5.0.2]
 - Discord.js Version: [13.0.1]

**Additional context**
It seems like _handleAutoplay() is not actually being called correctly (context of `this` switches since we are using it from within an event callback). Binding `this` within the connection.on("finish") callback seems to fix this, however even when actually being called, autoplay still doesn't function at all.

**Comment by joshuayuen99** [comment]
Correction: it does work with the above binding fix when trying to autoplay after a search phrase track, however it still does not properly function with direct YouTube linked tracks (or others, but my understanding is that it's only intended to work with sources from YouTube).

**Comment by joshuayuen99** [comment]
This seems to only be an issue when using https://github.com/DevSnowflake/discord-player-downloader along with `discord-player`. I'll open up an issue over there instead.

**Comment by DevAndromeda** [comment]
@joshuayuen99 ok so after doing some research, I've found the reason for this. When you have `@discord-player/downloader`, discord-player first passes all of your queries to the downloader and if the downloader resolves your query, it proceeds to play else it passes the query to rest of the extractors. And since `downloader` is a youtube-dl instance, it resolves almost all queries and the source type from downloader is not `youtube`, instead `arbitrary`. And `autoplay` supports `youtube` only.

**Comment by joshuayuen99** [comment]
@DevAndromeda I see. It does seem like youtube-dl is capable of correctly determining what the source type is, as I mentioned in https://github.com/DevSnowflake/discord-player-downloader/issues/4#issue-964452345.

I messed around with making the proposed changes (passing in `info.extractor` as an additional `source` property from `@discord-player/downloader`) and editing `ExtractorModel.ts` to support taking in the additional `source` property, which seemed to work fine (would correctly autoplay songs from direct YouTube links), however I began running into some issues when calling stop() after playing a YouTube link and enabling autoplay:

```
E:\TundraBot\node_modules\@discordjs\voice\dist\util\entersState.js:17
        const timeout = setTimeout(() => reject(new Error(`Did not enter state ${status} within ${maxTime}ms`)), maxTime);
                                                ^

Error: Did not enter state ready within 20000ms
    at Timeout.<anonymous> (E:\discordbot\node_modules\@discordjs\voice\dist\util\entersState.js:17:49)
    at listOnTimeout (node:internal/timers:557:17)
    at processTimers (node:internal/timers:500:7)
```

After some more experimentation, I found I get this error with the unedited version of `discord-player` and `@discord-player/downloader` as well (when playing a song via search phrase, resulting in the correct `youtube` source), enabling autoplay, and calling stop().

Another error that I also ran into instead of the above, that I unfortunately didn't grab the stack trace of, was something along the lines of not being able to use a destroyed stream. I'll open a new issue for this problem.
